### DISCUSSION TITLE
How to handle multiple subapps?

### DISCUSSION THREAD

**Discussion by lucascebertin** [talentlessguy/tinyhttp#276] [Q&A]
Hey, last few days I came with a question about how to handle "separated routes" and just started to using subapps for that.
So, after that I started adding more subapps and I have one that gives me a swagger docs.

My main app looks like this:
```typescript
const app = new App()
app.use(swaggerMiddleware) //just because I want all requests pass through this
app.use('/myroute', otherSubAppWithAnotherRoutes)
```

My `swaggerMiddleware`:
```typescript
const swaggerSubApp = new App()
swaggerSubApp.use(functionThatGeneratesTheSwaggerHandler(parametersForSwagger))
export default swaggerSubApp
```

As you can see I have two "nested .use(...)" and when the first `.use()` is invoked, doesnt matter if I call next() at the end of `functionThatGeneratesTheSwaggerHandler`, my next SubApp is never invoked.

I'm trying to understand if this is how the thing is meant to be, if I'm doing something (completely) wrong here or something else. 
For this very moment I stopped using subapps, just passing the main app to some functions to set up the routes.

Thanks in advance.

**Comment by v1rtl** [comment]
hello,

so the issue is that nested subapps don't work?

**Reply by lucascebertin** [reply]
That is what looks like to me but can't say for sure

**Reply by v1rtl** [reply]
@lucascebertin got it, thanks

tomorrow I'll check if it occurs in a bare repo and if it does I'll raise an issue

sorry for inconvenience :/

**Reply by lucascebertin** [reply]
Hey, don't worry at all about it.
I just appreciate all the hard work and your time trying to understand and help me.

**Reply by lucascebertin** [reply]
I've created a repo to reproduce this, you can check this out if needed:
https://github.com/lucascebertin/tinyhttp-multiple-subapps

**Reply by v1rtl** [reply]
here's a smaller repro:

```js
import { App } from '@tinyhttp/app'

const app = new App(), app1 = new App(), app2 = new App()

app2.use((req, res, next) => {
	console.log('ping')
	res.send('hi from app 2')
})

app1.use((req, res, next) => {
	console.log('pong')
	next()
})

app.use('/test', app1)
app.use('/test', app2)


app.listen(3000)
```

bug is confirmed, opening an issue (#279)

**Comment by Doc999tor** [comment]
I prepare even smaller and simpler reproducible example
http://localhost:3000/ping - works correctly 
http://localhost:3000/pong - returns 404 no matter what

```javascript
import { App } from '@tinyhttp/app'
const app = new App(), app1 = new App(), app2 = new App()

app1.get('/ping', (req, res, next) => {
	res.send('ping')
})

app2.get('/pong', (req, res, next) => {
	res.send('pong')
})

app.use(app1)
app.use(app2)

app.listen(3000)

```
